ion/cmp_test.go

```diff
@@ -0,0 +1,229 @@
+package ion
+
+import (
+	"math/big"
+	"reflect"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+type ionEqual interface {
+	eq(other ionEqual) bool
+}
+
+type ionFloat float64
+type ionInt int
+type ionInt64 int64
+type ionUint64 uint64
+type ionBigInt struct{ *big.Int }
+type ionBool bool
+type ionString string
+type ionTimestamp struct{ time.Time }
+type ionDecimal Decimal
```

Comment:
Per our offline conversation, I believe you should be able to define `ionEqual` for the built-in types (`float64`, `uint64`, etc) without having to create wrappers.

Reply:
Unfortunately we cannot define new methods to built-in types: `"Cannot define new methods on non-local type 'builtin.bool'"`.
More info on this: [Go Github issue](https://github.com/golang/go/issues/31631)

Possible solutions to define new methods are by 1)Type definition or 2)Embedding ([code snippet example](https://play.golang.org/p/JJnPrFZZXs0) / [stack overflow](https://stackoverflow.com/questions/28800672/how-to-add-new-methods-to-an-existing-type-in-go))